platform_channels/test/src/method_channel_demo_test.dart

```diff
@@ -8,38 +8,36 @@ import 'package:flutter_test/flutter_test.dart';
 import 'package:platform_channels/src/method_channel_demo.dart';
 
 void main() {
-  TestWidgetsFlutterBinding.ensureInitialized();
-
-  final methodChannel = MethodChannel('methodChannelDemo');
-
-  // Register a mock MethodCallHandler.
-  methodChannel.setMockMethodCallHandler((call) async {
-    var count = call.arguments['count'] as int;
-    if (call.method == 'increment') {
-      return ++count;
-    } else if (call.method == 'decrement') {
-      return --count;
-    }
-
-    return MissingPluginException();
-  });
-
   group('MethodChannelDemo tests', () {
+    final methodChannel = MethodChannel('methodChannelDemo');
+
     testWidgets('MethodChannelDemo count test', (tester) async {
       await tester.pumpWidget(MaterialApp(
         home: MethodChannelDemo(),
       ));
 
+      // Register a mock MethodCallHandler.
```

Comment:
There are a number of places you can put this sort of setup code. Inside at the top of main is fine, and there's also a `setUpAll` method that you can use to put code that needs to be called once before all the tests. The only thing I'd really recommend is to keep the declaration of `methodChannel` and this assignment in the same spot in the code, since they're directly related to one another.

Reply:
@RedBrogdon I moved the code to register `MethodCallHandler` inside the `testWidgets` because I wanted to make sure I was accessing the `ServicesBinding.defaultBinaryMessenger` after the binding was initialized in `testWidgets`. If I register the mock in main, I had to use `TestWidgetsFlutterBinding.ensureInitialized`.

But, that's a good suggestion, I can use `setUpAll`.

Reply:
I'll do the required changes; I'll use the `setUp` in the group to register the mock for `MethodCallHandler`.

Reply:
Got it. You're right, that makes total sense. `setUpAll` should prevent you from any similar problems in the future (at least in this file).